ustriage/ustriage.py

```diff
@@ -311,9 +311,25 @@ def handle_webbrowser(open_in_browser, url):
         time.sleep(5)
 
 
+# 7-bit C1 ANSI sequences
+ANSI_ESCAPE = re.compile(r'''
+    \x1B  # ESC
+    (?:   # 7-bit C1 Fe (except CSI)
+        [@-Z\\-_]
+    |     # or [ for CSI, followed by a control sequence
+        \[
+        [0-?]*  # Parameter bytes
+        [ -/]*  # Intermediate bytes
+        [@-~]   # Final byte
+    )
+''', re.VERBOSE)
```

Comment:
I have never used the VERBOSE flag before. interestingly, docs say that "Whitespace within the pattern is ignored, except when **...** or within tokens like *?, **(?:** or (?P<...>. Which is indeed the case here. What is also interesting is that the regular expression does work as intended (as far as I could test it). Maybe I misinterpreted it somehow?

Reply:
Let me be honest, I have used this from examples I've found and after three other solutions failing this was the good one I've taken over and adapted as-is. But yes after I've found it it just makes the regex more readable. So I liked and prefer it this way. Be careful that the rule says "within" ```(?:``` which we do not have - we only have after/before and that is ignored.

better:

```
    \x1B  # ESC
    (?:   # 7-bit C1 Fe (except CSI)
        [@-Z\\-_]
    |     # or [ for CSI, followed by a control sequence
        \[
        [0-?]*  # Parameter bytes
        [ -/]*  # Intermediate bytes
        [@-~]   # Final byte
    )
```

vs

```r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])'```

Reply:
> Be careful that the rule says "within" (?: which we do not have - we only have after/before and that is ignored.

When it says within `(?:`, I believed it was something in `(?: here )`, mostly because it refers to `(?:` as a token.
Out of curiosity, I went through cpython and found https://github.com/python/cpython/pull/4366, which is explained in https://github.com/python/cpython/issues/59811. It makes it clear that the issue is with constructs such as `(? : a | b )` (opposing constructs such as `(?: a | b )`.